packages/gatsby-plugin-page-creator/src/collection-extract-query-string.ts

```diff
@@ -0,0 +1,54 @@
+import { babelParseToAst } from "gatsby/dist/utils/babel-parse-to-ast"
+import { generateQueryFromString } from "./extract-query"
+import { getGraphQLTag } from "babel-plugin-remove-graphql-queries"
+import fs from "fs-extra"
+import traverse from "@babel/traverse"
+
+// This Function opens up the actual collection file and extracts the queryString used in the
+// `unstable_createPagesFromData` macro.
+export function collectionExtractQueryString(
+  absolutePath: string
+): string | null {
+  let queryString: string | null = null
+
+  // 1.  Convert the file to a babel ast
+  const ast = babelParseToAst(
+    fs.readFileSync(absolutePath).toString(),
+    absolutePath
+  )
+
+  const modelType = /\{([a-zA-Z]+)\./.exec(absolutePath)?.[1]
+
+  if (!modelType) {
+    throw new Error(`You screwed up`)
```

Comment:
I just got this error when testing 😅

Tested with the old naming convention, so basically, it failed with `{fields__slug}.js`. I feel that we need a message here explaining the expected name format.

Reply:
Haha yeah. I need to improve that. It was a quick thing :)